contract/src/lib.rs

```diff
@@ -216,6 +216,9 @@ impl MpcContract {
                 if !participants.contains_key(&acc_id_to_leave) {
                     env::panic_str("account to leave is not in the participant set");
                 }
+                if participants.len() <= *threshold {
+                    env::panic_str("the number of participants can not go below the threshold");
+                }
```

Comment:
Let's say, we have 5 nodes and T=4.
- 4 nodes vote for n5 to leave, success
- 4 remaining nodes vote for n4 to leave (including n4). `voted.len() >= *threshold` allows that, success
- we are ending up with 3 nodes and T4

The check that I have added (`participants.len() <= *threshold`) will prevent step 2 from happening.

Reply:
Then to be more concise here, we should realistically be having a check of `participants.len == threshold` since going below should never be a possibility. if it is, then this check still would be meaningless

Reply:
I see 0 benefits in the making it `==` instead of `<=`. But it makes it less bulletproof.
But I agree, we should never go below. We can make sure of that with more testing later.